src/backend/distributed/deparser/deparse_function_stmts.c

```diff
@@ -274,25 +274,27 @@ AppendDefElemSet(StringInfo buf, DefElem *def)
 	{
 		case VAR_SET_VALUE:
 		{
-			appendStringInfo(buf, " SET %s = %s", setStmt->name, setVariableArgs);
+			appendStringInfo(buf, " SET %s = %s", quote_identifier(setStmt->name),
```

Comment:
Can we add a test case for this?

Note that you can always SET prefix.whatever to something and read it using current_setting.